File: react-frontend/src/services/inventoryService.ts
import { http } from './http';
import type { ApiResponse } from '../types';

export interface InventoryItem {
  id: number;
  branchId: number;
  name: string;
  unit: string;
  quantity: number;
  minStock: number;
  stockStatus: 'SUFFICIENT' | 'LOW' | 'OUT_OF_STOCK';
  expiryDate?: string;
  updatedAt: string;
  createdAt: string;
}

export interface InventoryTransaction {
  id: number;
  inventoryItemId: number;
  type: 'IMPORT' | 'EXPORT';
  quantity: number;
  reason?: string;
  note?: string;
  performedBy: number;
  performedByName: string;
  createdAt: string;
}

export const inventoryService = {
  getAll: (branchId: number, search = '') =>
    http.get<ApiResponse<InventoryItem[]>>(
      `/inventory/branch/${branchId}?search=${encodeURIComponent(search)}`
    ),

  create: (branchId: number, data: { name: string; unit: string; quantity: number; minStock: number; expiryDate?: string }) =>
    http.post<ApiResponse<InventoryItem>>(`/inventory/branch/${branchId}`, data),

  update: (id: number, data: { name: string; unit: string; minStock: number; expiryDate?: string }) =>
    http.put<ApiResponse<InventoryItem>>(`/inventory/${id}`, data),

  delete: (id: number) =>
    http.delete<ApiResponse<null>>(`/inventory/${id}`),

  importStock: (id: number, data: { quantity: number; note?: string; performedBy: number; performedByName: string }) =>
    http.post<ApiResponse<InventoryItem>>(`/inventory/${id}/import`, data),

  exportStock: (id: number, data: { quantity: number; reason: string; note?: string; performedBy: number; performedByName: string }) =>
    http.post<ApiResponse<InventoryItem>>(`/inventory/${id}/export`, data),

  getTransactions: (id: number) =>
    http.get<ApiResponse<InventoryTransaction[]>>(`/inventory/${id}/transactions`),
};

File: backend/src/main/java/com/ioc/internship/service/InventoryService.java
package com.ioc.internship.service;

import com.ioc.internship.dto.InventoryItemDTO;
import com.ioc.internship.dto.InventoryTransactionDTO;

import java.util.List;

public interface InventoryService {
    List<InventoryItemDTO> getInventoryByBranch(Long branchId, String search);
    InventoryItemDTO createItem(Long branchId, String name, String unit, Double quantity, Double minStock);
    InventoryItemDTO updateItem(Long id, String name, String unit, Double minStock);
    void deleteItem(Long id);

    InventoryItemDTO importStock(Long id, Double quantity, String note, Long performedBy, String performedByName);
    InventoryItemDTO exportStock(Long id, Double quantity, String reason, String note, Long performedBy, String performedByName);

    List<InventoryTransactionDTO> getTransactions(Long inventoryItemId);
}
